OutOfSchool/OutOfSchool.WebApi.Tests/Services/ApplicationServiceTests.cs

```diff
@@ -439,6 +442,42 @@ public async Task UpdateApplication_WhenIdIsValid_ShouldReturnApplication()
         AssertApplicationsDTOsAreEqual(expected, result);
     }
 
+    [Test]
+    public async Task UpdateApplication_WhenIdIsValidAndNeedUpdateWorkshopStatus_ShouldReturnApplication()
+    {
+        // Arrange
+        var id = new Guid("1745d16a-6181-43d7-97d0-a1d6cc34a8bd");
+        var entity = WithApplication(id);
+        var changedEntity = WithApplication(id, ApplicationStatus.Approved);
+        var userId = Guid.NewGuid().ToString();
+        var workshop = WithWorkshop(new Guid("0083633f-4e5b-4c09-a89d-52d8a9b89cdb"));
+
+        var applicationsMock = WithApplicationsList().AsQueryable().BuildMock();
+
+        applicationRepositoryMock.Setup(r => r.Get(
+                It.IsAny<int>(),
+                It.IsAny<int>(),
+                It.IsAny<string>(),
+                It.IsAny<Expression<Func<Application, bool>>>(),
+                It.IsAny<Dictionary<Expression<Func<Application, object>>, SortDirection>>(),
+                It.IsAny<bool>()))
+            .Returns(applicationsMock)
+            .Verifiable();
+
+        applicationRepositoryMock.Setup(a => a.Update(It.IsAny<Application>(), It.IsAny<Action<Application>>())).ReturnsAsync(changedEntity);
+        applicationRepositoryMock.Setup(a => a.GetById(It.IsAny<Guid>())).ReturnsAsync(entity);
+        applicationRepositoryMock.Setup(a => a.Count(x => x.WorkshopId == workshop.Id && (x.Status == ApplicationStatus.Approved || x.Status == ApplicationStatus.StudyingForYears))).ReturnsAsync(1);
```

Comment:
The same

Reply:
See above...

Reply:
You are right. setup the Count method is not needed (because returned result always equal 0).